### ISSUE TITLE
Add support for directory based session

### ISSUE THREAD

**Issue by kebot** [xolox/vim-session#49]
for example, if I'm at
    ~/projects/first_project/

vim-session will auto load session from ~/projects/first_project/.vimsession

**Comment by Paschalis** [comment]
I am having issues with the proposed solutions.
I've adapted the above approach to use a different directory, `.sessions`, so it will be hidden.
Relevant lines from vimrc:
```
let s:session_base_directory = g:PATH_RT ."/.sessions"
let g:session_directory = xolox#misc#path#merge(s:session_base_directory, getcwd())
```
An example of `.sessions/` with this approach is:

```
<vim runtime>
<vim runtime>/.sessions/
<vim runtime>/.sessions/User/
<vim runtime>/.sessions/User/USERNAME/
<vim runtime>/.sessions/User/USERNAME/default.vim
<vim runtime>/.sessions/User/USERNAME/Documents/
<vim runtime>/.sessions/User/USERNAME/Documents/default.vim
```

It creates the folders as expected for each new session in `<vim-runtime>/.sessions`.
However, there are cases when a session does not exist and it still restores some buffers.
It's like it picks up some other sessions to restore from. 
Also, when I am trying to open a new file, e.g. `vim test`, where `test` does not exist yet, it ignores it, and loads instead the session of the directory.
Normally, `xolox#session#is_empty()` should have returned true in this case.

BTW this approach creates lots of folders in`.session`. To make it "cleaner", I've used the following code, which changes the filename instead of the directory:

```
let g:session_directory = g:PATH_RT ."/.sessions"
let g:session_default_name= substitute(getcwd(),"/","_", "g") . '.session'
```

An example of `.sessions` with this approach:

```
<vim runtime>
<vim runtime>/.sessions/
<vim runtime>/.sessions/User_USERNAME.session.vim
<vim runtime>/.sessions/User_USERNAME_Documents.session.vim
```

This approach still has the above issues though.

An alternative solution would be to create a session file in each folder we are working. There is a closed issue that describes this, but I personally don't like this approach. I'd like to keep things clean, and put all sessions in a single folder.

So any ideas where the issue might be? I'll gladly provide more information if asked to.
BTW thanks for your great plugins!
